File: LeetCode/src/1603.rs
struct ParkingSystem {
    big: i32,
    medium: i32,
    small: i32,
}

impl ParkingSystem {
    fn new(big: i32, medium: i32, small: i32) -> Self {
        Self { big, medium, small }
    }

    fn add_car(&mut self, car_type: i32) -> bool {
        if car_type == 1 && self.big > 0 {
            self.big -= 1;
            true
        } else if car_type == 2 && self.medium > 0 {
            self.medium -= 1;
            true
        } else if car_type == 3 && self.small > 0 {
            self.small -= 1;
            true
        } else {
            false
        }
    }
}

File: Go/parking_system/1603.go
/**
 * Runtime: 35ms (Beats 87.33%)
 * Memory: 7.8MB (Beats 32.67%)
 */

package parkingsystem

type ParkingSystem struct {
	small  int
	medium int
	big    int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{small, medium, big}
}

func (p *ParkingSystem) AddCar(carType int) bool {
	if carType == 1 && p.big > 0 {
		p.big--
		return true
	}
	if carType == 2 && p.medium > 0 {
		p.medium--
		return true
	}
	if carType == 3 && p.small > 0 {
		p.small--
		return true
	}
	return false
}
